linalg/vector.hpp

```diff
@@ -121,7 +121,7 @@ class Vector
       : data(size_, h_mt, d_mt), size(size_) { }
 
    /// Create a vector from a statically sized C-style array of convertible type
-   template <int N, typename CT>
+   template <typename CT, int N>
```

Comment:
This is not exactly backward-compatible -- e.g. when using explicit template parameters.

Reply:
True, but I suspect none of our users have explicitly specified them. We can revert the order if you want though. I changed it for consistency with Array, but it's not so important.

Reply:
Yes, it is not very likely that someone used explicit template parameters. I'll leave it up to you to decide if you want to revert -- I'm fine either way.